tests/components/ipma/test_config_flow.py

```diff
@@ -1,116 +1,57 @@
 
-    result = await flow.async_step_user()
     assert result["type"] == "form"
     assert result["step_id"] == "user"
 
 
-async def test_flow_show_form() -> None:
-    """Test show form scenarios first time.
+async def test_flow_entry_already_exists(hass: HomeAssistant) -> None:
+    """Test user input for config_entry that already exists.
 
-    Test when the form should show when no configurations exists
+    Test when the form should show when user puts existing location
+    in the config gui. Then the form should show with error.
     """
-    hass = Mock()
-    flow = config_flow.IpmaFlowHandler()
-    flow.hass = hass
-
-    with patch(
-        "homeassistant.components.ipma.config_flow.IpmaFlowHandler._show_config_form"
-    ) as config_form:
-        await flow.async_step_user()
-        assert len(config_form.mock_calls) == 1
-
-
-async def test_flow_entry_created_from_user_input() -> None:
-    """Test that create data from user input.
-
-    Test when the form should show when no configurations exists
-    """
-    hass = Mock()
-    flow = config_flow.IpmaFlowHandler()
-    flow.hass = hass
-
-    test_data = {"name": "home", CONF_LONGITUDE: "0", CONF_LATITUDE: "0"}
-
-    # Test that entry created when user_input name not exists
-    with patch(
-        "homeassistant.components.ipma.config_flow.IpmaFlowHandler._show_config_form"
-    ) as config_form, patch.object(
-        flow.hass.config_entries,
-        "async_entries",
-        return_value=[],
-    ) as config_entries:
-        result = await flow.async_step_user(user_input=test_data)
-
-        assert result["type"] == "create_entry"
-        assert result["data"] == test_data
-        assert len(config_entries.mock_calls) == 1
-        assert not config_form.mock_calls
-
-
-async def test_flow_entry_config_entry_already_exists() -> None:
-    """Test that create data from user input and config_entry already exists.
-
-    Test when the form should show when user puts existing name
-    in the config gui. Then the form should show with error
-    """
-    hass = Mock()
-    flow = config_flow.IpmaFlowHandler()
-    flow.hass = hass
-
-    test_data = {"name": "home", CONF_LONGITUDE: "0", CONF_LATITUDE: "0"}
-
-    # Test that entry created when user_input name not exists
-    with patch(
-        "homeassistant.components.ipma.config_flow.IpmaFlowHandler._show_config_form"
-    ) as config_form, patch.object(
-        flow.hass.config_entries, "async_entries", return_value={"home": test_data}
-    ) as config_entries:
-        await flow.async_step_user(user_input=test_data)
-
-        assert len(config_form.mock_calls) == 1
-        assert len(config_entries.mock_calls) == 1
-        assert len(flow._errors) == 1
+    test_data = {
+        CONF_NAME: "home",
+        CONF_LONGITUDE: 0,
+        CONF_LATITUDE: 0,
+    }
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=test_data
+    )
```

Comment:
Create a mock config entry with the longitude + latitude = 0 and add it to hass.

Reply:
done